torch/_dynamo/eval_frame.py

```diff
@@ -262,19 +264,125 @@ def innermost_fn(fn):
     return unaltered_fn
 
 
+dynamo_guarded_config_ignorelist = {
+    "log_file_name",
+    "verbose",
+    "verify_correctness",  # will not affect model, will raise RuntimeError
+    # (no silent change to compilation behaviour)
+    "cache_size_limit",
+    "accumulated_cache_size_limit",
+    "print_specializations",
+    "replay_record_enabled",
+    "cprofile",  # only wraps _compile, not graph
+    "repro_after",
+    "repro_level",
+    "repro_forward_only",
+    "repro_tolerance",
+    "same_two_models_use_fp64",
+    "error_on_recompile",  # safe because: will throw error
+    "report_guard_failures",
+    "report_all_guard_failures",
+    "base_dir",  # used for minifying / logging
+    "translation_validation",
+    "translation_validation_timeout",
+    "translation_validation_no_bisect",
+    "DEBUG_DIR_VAR_NAME",
+    "debug_dir_root",
+}
+
+dynamo_guarded_config_string_serialization_list = {
+    # "is_fbcode",  # function def (removed by ConfigModule visit)
+    "skipfiles_inline_module_allowlist",  # cannot pickle module (unused)
+    "constant_functions",  # PyCapsules (i.e. function ptrs)
+    "traceable_tensor_subclasses",  # Cannot pickle local object
+}
+
+# The config to restore to should dynamo compile / recompile when
+# executing from the compiled function's _TorchDynamoContext
+config_cache = threading.local()
+
+
+def _maybe_init_guarded_config_cache():
+    if hasattr(config_cache, "saved_config"):
+        return
+    else:
+        for c in ["saved_config", "saved_config_hash", "cached_recompile_hash"]:
```

Comment:
can you doc what each of these mean

Reply:
let me just combine into tuple `"saved_config_and_hash"` and move cached_recompile_hash into `ConfigModule._cached_config_hash`